xmipptomo/protocols/protocol_deep_misalignment_detection.py

```diff
@@ -0,0 +1,426 @@
+
+        return overallPrediction, predictionAverage
+
+    def getOutputSetOfAlignedTomograms(self):
+        if self.alignedTomograms:
+            self.alignedTomograms.enableAppend()
+
+        else:
+            alignedTomograms = self._createSetOfTomograms(suffix='Ali')
+
+            alignedTomograms.copyInfo(self.isot)
+
+            alignedTomograms.setStreamState(Set.STREAM_OPEN)
+
+            self._defineOutputs(alignedTomograms=alignedTomograms)
+            self._defineSourceRelation(self.isot, alignedTomograms)
+
+        return self.alignedTomograms
+
+    def getOutputSetOfStrongMisalignedTomograms(self):
+        if self.strongMisalignedTomograms:
+            self.strongMisalignedTomograms.enableAppend()
+
+        else:
+            strongMisalignedTomograms = self._createSetOfTomograms(suffix='StrongMisali')
+
+            strongMisalignedTomograms.copyInfo(self.isot)
+
+            strongMisalignedTomograms.setStreamState(Set.STREAM_OPEN)
+
+            self._defineOutputs(strongMisalignedTomograms=strongMisalignedTomograms)
+            self._defineSourceRelation(self.isot, strongMisalignedTomograms)
+
+        return self.strongMisalignedTomograms
+
+    def getOutputSetOfWeakMisalignedTomograms(self):
+        if self.weakMisalignedTomograms:
+            self.weakMisalignedTomograms.enableAppend()
+
+        else:
+            weakMisalignedTomograms = self._createSetOfTomograms(suffix='WeakMisali')
+
+            weakMisalignedTomograms.copyInfo(self.isot)
+
+            weakMisalignedTomograms.setStreamState(Set.STREAM_OPEN)
+
+            self._defineOutputs(weakMisalignedTomograms=weakMisalignedTomograms)
+            self._defineSourceRelation(self.isot, weakMisalignedTomograms)
+
+        return self.weakMisalignedTomograms
+
+    def getOutputSetOfSubtomos(self):
+        if self.outputSubtomos:
+            self.outputSubtomos.enableAppend()
+
+        else:
+            outputSubtomos = self._createSetOfSubTomograms(suffix="FS")
+
+            outputSubtomos.copyInfo(self.isot)
+            outputSubtomos.setSamplingRate(TARGET_SAMPLING_RATE)
+
+            outputSubtomos.setStreamState(Set.STREAM_OPEN)
+
+            self._defineOutputs(outputSubtomos=outputSubtomos)
+            self._defineSourceRelation(self.isot, outputSubtomos)
+
+        return self.outputSubtomos
+
+    @staticmethod
+    def getSubtomoPathList(coordFilePath):
+        coordFilePath_noExt = os.path.splitext(coordFilePath)[0]
+        counter = 1
+
+        subtomoPathList = []
+
+        while True:
+            subtomoPath = coordFilePath_noExt + '-' + str(counter) + '.mrc'
+
+            if not os.path.exists(subtomoPath):
+                break
+
+            subtomoPathList.append(subtomoPath)
+            counter += 1
+
+        return subtomoPathList
+
+    # --------------------------- INFO functions ----------------------------
+    def _summary(self):
+        summary = []
```

Comment:
Could you provide any stadistic to know about the fiducial alignment?

Reply:
done